src/components/Transcript/TranscriptSection.ts

```diff
@@ -88,23 +89,45 @@ export class TranscriptSection implements RedomComponent {
     }
   };
 
+  private createSpeakerElements = () => {
+    const { speaker } = this.paragraph;
+    const firstname = speaker.unknown ? 'Mluvčí' : `${speaker.firstname ?? ''}` as string;
+    const surname = speaker.unknown ? '' : speaker.surname as string;
+    const role = speaker.unknown || speaker.role === undefined ? '' : speaker.role as string;
```

Comment:
From the Speaker type it is clear that role cannot be undefined. Also unnecessary typecasting